app/javascript/history/history.jsx

```diff
@@ -0,0 +1,212 @@
+import { h, Component } from 'preact';
+import { PropTypes } from 'preact-compat';
+import debounce from 'lodash.debounce';
+import setupAlgoliaIndex from '../src/utils/algolia';
+
+export class History extends Component {
+  constructor(props) {
+    super(props);
+
+    const { availableTags } = this.props;
+    this.state = {
+      query: '',
+      items: [],
+      totalCount: 0,
+      index: null,
+      itemsLoaded: false,
+      hitsPerPage: 100,
+      availableTags,
+      selectedTags: [],
+      page: 0,
+      showNextPageButton: false,
+    };
+
+    this.handleTyping = debounce(this.handleTyping.bind(this), 300, {
+      leading: true,
+    });
+  }
+
+  componentDidMount() {
+    const index = setupAlgoliaIndex({
```

Comment:
I plan to use this utility function in the future, because I've noticed the algolia init code is repeated everywhere